### ISSUE TITLE
Basic Oven features are not working

### ISSUE THREAD

**Issue by fotisbaco** [rowysock/home-assistant-HomeWhiz#76]
After upgrade to v0.2.2, oven discovered but sensors not accessed

Controls:

Booster
Programme
Skip
State
Temperature

Sensors:

Home Oven Cooking duration (Hour : Minute)
Home Oven Programme End Time
Home Oven Remaining
Home Oven Sub-state
Home Oven Temperature

Thanks for great job!

![oven_1](https://user-images.githubusercontent.com/60899404/209688638-a3a717f9-5d48-47c9-ba0e-b99ba81f397d.jpg)

**Comment by rowysock** [comment]
@fotisbaco The controls and sensors you have shown are the basic ones, that should work already. Can you share the logs?

**Comment by TechHummel** [comment]
You could also try to change some oven settings and check if the sensors are updaing then!

**Comment by fotisbaco** [comment]
> @fotisbaco The controls and sensors you have shown are the basic ones, that should work already. Can you share the logs?

Because i don't see any error in log's, could you tell me how to find any info to help you. I'm enable the logger

**Comment by fotisbaco** [comment]
> You could also try to change some oven settings and check if the sensors are updaing then!

I do but nothing change

**Comment by rowysock** [comment]
@fotisbaco Please enable the debug logging as described here: https://github.com/rowysock/home-assistant-HomeWhiz#retrieve-integration-logs
Then Reload this integration, and collect the logs. It should look something like this
```
2022-12-27 19:58:50.411 INFO (MainThread) [custom_components.homewhiz] Setting up entry T999902890777401659617
2022-12-27 19:58:50.412 INFO (MainThread) [custom_components.homewhiz] Setting up cloud connection
2022-12-27 19:58:50.431 DEBUG (MainThread) [custom_components.homewhiz] Appliance is AC, not adding Select entities
2022-12-27 19:58:50.434 DEBUG (MainThread) [custom_components.homewhiz] Appliance is AC, not adding Sensor entities
2022-12-27 19:58:50.439 INFO (MainThread) [custom_components.homewhiz] Connecting to T999902890777401659617
2022-12-27 19:58:50.971 DEBUG (MainThread) [custom_components.homewhiz] Credentials expire at: 2022-12-27 20:58:50
2022-12-27 19:58:51.661 DEBUG (Dummy-3) [custom_components.homewhiz] Message received: bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06\x15\x03\x00\x00\x00\x1c\x00\x00\x14\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
2022-12-27 19:58:51.662 DEBUG (Dummy-3) [custom_components.homewhiz] Manually updated homewhiz data

```

**Comment by fotisbaco** [comment]
Hi,

I hope this help

2022-12-27 22:50:48.810 INFO (MainThread) [custom_components.homewhiz] Setting up entry F999904668828549174231
2022-12-27 22:50:48.810 INFO (MainThread) [custom_components.homewhiz] Setting up cloud connection
2022-12-27 22:50:48.814 INFO (MainThread) [homeassistant.components.select] Setting up select.homewhiz
2022-12-27 22:50:48.815 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.homewhiz
2022-12-27 22:50:48.815 INFO (MainThread) [homeassistant.components.climate] Setting up climate.homewhiz
2022-12-27 22:50:48.948 DEBUG (MainThread) [custom_components.homewhiz] Generating descriptions from config
2022-12-27 22:50:48.948 DEBUG (MainThread) [custom_components.homewhiz] Adding STATE EnumEntityDescription
2022-12-27 22:50:48.949 DEBUG (MainThread) [custom_components.homewhiz] Selects: ['STATE', 'OVEN_PROGRAM', 'OVEN_TEMPERATURE', 'OVEN_BOOSTER', 'OVEN_SKIP_PREHEATING']
2022-12-27 22:50:49.070 DEBUG (MainThread) [custom_components.homewhiz] Generating descriptions from config
2022-12-27 22:50:49.070 DEBUG (MainThread) [custom_components.homewhiz] Adding SUB_STATE EnumEntityDescription
2022-12-27 22:50:49.070 DEBUG (MainThread) [custom_components.homewhiz] Adding config progress variables
2022-12-27 22:50:49.070 DEBUG (MainThread) [custom_components.homewhiz] Sensors: ['SUB_STATE', 'TEMPERATURE', 'VARIABLE_DELAY', 'VARIABLE_OVEN_DURATION', 'VARIABLE_REMAINING']
2022-12-27 22:50:49.189 DEBUG (MainThread) [custom_components.homewhiz] Appliance is not AC, not adding Climate entity

**Comment by rowysock** [comment]
This is all? there are no logs after that? 
We should see a line like 
`2022-12-27 19:58:50.439 INFO (MainThread) [custom_components.homewhiz] Connecting to T999902890777401659617`

**Comment by fotisbaco** [comment]
Hi,

I'm uninstall and install again the integration, and upload new logs

2022-12-28 10:51:50.566 INFO (MainThread) [custom_components.homewhiz] Setting up entry F999904668828549174231
2022-12-28 10:51:50.566 INFO (MainThread) [custom_components.homewhiz] Setting up cloud connection
2022-12-28 10:51:50.571 INFO (MainThread) [homeassistant.components.select] Setting up select.homewhiz
2022-12-28 10:51:50.571 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.homewhiz
2022-12-28 10:51:50.571 INFO (MainThread) [homeassistant.components.climate] Setting up climate.homewhiz
2022-12-28 10:51:50.571 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [139956597066336] Sending [{"id":138,"type":"event","event":[{"type":"updated","entry":{"entry_id":"9a22e0891468676fa8d5c24dd871a298","domain":"homewhiz","title":"Home Oven","source":"user","state":"setup_in_progress","supports_options":false,"supports_remove_device":false,"supports_unload":true,"pref_disable_new_entities":false,"pref_disable_polling":false,"disabled_by":null,"reason":null}}]},{"id":138,"type":"event","event":[{"type":"updated","entry":{"entry_id":"9a22e0891468676fa8d5c24dd871a298","domain":"homewhiz","title":"Home Oven","source":"user","state":"setup_in_progress","supports_options":false,"supports_remove_device":false,"supports_unload":true,"pref_disable_new_entities":false,"pref_disable_polling":false,"disabled_by":null,"reason":null}}]}]
2022-12-28 10:51:50.624 DEBUG (MainThread) [custom_components.homewhiz] Generating descriptions from config
2022-12-28 10:51:50.624 DEBUG (MainThread) [custom_components.homewhiz] Adding STATE EnumEntityDescription
2022-12-28 10:51:50.624 DEBUG (MainThread) [custom_components.homewhiz] Selects: ['STATE', 'OVEN_PROGRAM', 'OVEN_TEMPERATURE', 'OVEN_BOOSTER', 'OVEN_SKIP_PREHEATING']
2022-12-28 10:51:50.651 DEBUG (MainThread) [custom_components.homewhiz] Generating descriptions from config
2022-12-28 10:51:50.651 DEBUG (MainThread) [custom_components.homewhiz] Adding SUB_STATE EnumEntityDescription
2022-12-28 10:51:50.651 DEBUG (MainThread) [custom_components.homewhiz] Adding config progress variables
2022-12-28 10:51:50.651 DEBUG (MainThread) [custom_components.homewhiz] Sensors: ['SUB_STATE', 'TEMPERATURE', 'VARIABLE_DELAY', 'VARIABLE_OVEN_DURATION', 'VARIABLE_REMAINING']
2022-12-28 10:51:50.678 DEBUG (MainThread) [custom_components.homewhiz] Appliance is not AC, not adding Climate entity


2022-12-28 10:51:50.725 INFO (MainThread) [custom_components.homewhiz] Connecting to F999904668828549174231


2022-12-28 10:51:51.622 DEBUG (MainThread) [custom_components.homewhiz] Credentials expire at: 2022-12-28 11:51:51

**Comment by rowysock** [comment]
@fotisbaco I've released https://github.com/rowysock/home-assistant-HomeWhiz/releases/tag/v0.2.4 with a code that may fix your issue. It's really hard to tell why the integration is not working for you without access to your device/account :/

**Comment by fotisbaco** [comment]
> @fotisbaco I've released https://github.com/rowysock/home-assistant-HomeWhiz/releases/tag/v0.2.4 with a code that may fix your issue. It's really hard to tell why the integration is not working for you without access to your device/account :/

Not working.... It's ok for me to send you mail with gredentials to access my Homewhiz account

**Comment by rowysock** [comment]
Please do. My email is rowysock@gmail.com

**Comment by fotisbaco** [comment]
> Please do. My email is [rowysock@gmail.com](mailto:rowysock@gmail.com)

Check your mail please

**Comment by rowysock** [comment]
I've released the fixed code in v0.2.5. Please verify

**Comment by fotisbaco** [comment]
> I've released the fixed code in v0.2.5. Please verify

I confirm v0.2.5 solve the issue. Please close the case

Thanks
